### DISCUSSION TITLE
Error establishing database connection,

### DISCUSSION THREAD

**Discussion by cosc212101048** [EasyEngine/easyengine#1818] [Support Requests]
Hello 

My sites were working just fine but suddenly, all sites are giving "Error establishing database connection". 

After running `docker ps` 

I noticed that MariaDB container keeps restarting 
`6a6b484aaf56   easyengine/mariadb:v4.6.6       "docker-entrypoint.s…"   4 months ago   Restarting (1) 4 seconds ago`

I tried restarting everything, but i can't fix the issue

**Comment by petermann** [comment]
Hello @cosc212101048, 

I don't know if you've seen this post. Maybe it can help you: https://community.easyengine.io/t/how-to-solve-error-establishing-a-database-connection/14333

**Comment by sreekj** [comment]
Hello Team, I'm finding the same issue. Followed the link shared by petermann above. Freed up space. The issue persisted.
Found these in the docker logs of the db container:
2023-01-01 15:50:22 791 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).
2023-01-01 15:50:22 791 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'hist_type' at position 9 to have type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB','JSON_HB'), found type enum('SINGLE_PREC_HB','DOUBLE_PREC_HB').
2023-01-01 15:50:22 791 [ERROR] Incorrect definition of table mysql.column_stats: expected column 'histogram' at position 10 to have type longblob, found type varbinary(255).

Searching with these log entries in the internet lead me to adding these environment variables in the docker-compose.yml under the service global-db:
      - MARIADB_AUTO_UPGRADE=1
      - MARIADB_INITDB_SKIP_TZINFO=1
      
      
The global-db image and version in my /opt/easyengine/services/docker-compose is mariadb:v4.6.3

The issue still persists.

I'm guessing this might be what's happening to  @cosc212101048 too and your advices on it might help us both.

**Comment by petermann** [comment]
Hi,

Have you tried this procedure? You can find the instructions here: https://github.com/EasyEngine/easyengine/discussions/1759#discussioncomment-4505933

or try this command to check and repair tables

```
cd /opt/easyengine/services && docker-compose exec global-db bash -c 'mysqlcheck -uroot -p${MYSQL_ROOT_PASSWORD} --auto-repair --check --all-databases'
```
### Backup Recommendation:

**It is highly recommended to perform a complete backup before proceeding with any steps to ensure the safety of your data.**

Suggestion for a full backup (attention to file size):

```
cd /opt/easyengine/services && docker-compose exec global-db bash -c 'mysqldump -uroot -p${MYSQL_ROOT_PASSWORD} --all-databases' > ~/full-backup-db.sql && cd ~/ && tar -zcf full-backup-db.tar.gz full-backup-db.sql --remove-files && ls -l full-backup-db.tar.gz
```

**Reply by sreekj** [reply]
That worked flawlessly.
Thank you!

**Reply by sreekj** [reply]
for further clarity, I had tried: 
cd /opt/easyengine/services && docker-compose exec global-db bash -c 'mysqlcheck -uroot -p${MYSQL_ROOT_PASSWORD} --auto-repair --check --all-databases'

and it worked.

**Comment by marcoandrei** [comment]
@cosc212101048 did you resolve your problem with the orientations here?
